dot/core/service.go

```diff
@@ -523,7 +526,20 @@ func (s *Service) GetMetadata(bhash *common.Hash) ([]byte, error) {
 	}
 
 	rt, err := s.blockState.GetRuntime(bhash)
-	if err != nil {
+	if errors.Is(err, blocktree.ErrFailedToGetRuntime) {
+		rt, err = s.getRuntimeFromDB(*bhash)
+		if err != nil {
+			return nil, fmt.Errorf("getting runtime from database: %w", err)
+		}
+		metadata, err := rt.Metadata()
```

Comment:
in the outer scope, after we `GetRuntime` successfully we call `rt.SetContextStorage(ts)`, and here, in the inner scope we just call `rt.Metadata` directly, is there any reason?

Reply:
We don't set runtime context storage here because the runtime used from database is temporary and is closed as soon as metadata is found.